### ISSUE TITLE
last is broken

### ISSUE THREAD

**Issue by harendra-kumar** [composewell/streamly#9]
The implementation of `last` is broken if the stream ends abruptly using the `stop` continuation. In that case even though there may have been a last element we would not return it. Consider this:

```
    go m1 =
        let stop            = return Nothing
            yield a Nothing = return (Just a)
            yield _ (Just x) = go x
        in (runStream m1) Nothing stop yield
```

As soon as stop is called we return `Nothing` which is incorrect. We should remember and thread around the last element via `go` and in case `stop` is called we should yield the last one.

**Comment by harendra-kumar** [comment]
We need tests that include streams that stop abruptly using the `stop` continuation instead of using `yield a Nothing`.

**Comment by harendra-kumar** [comment]
Fixed by #25, thanks @Abhiroop !
